prow/spyglass/gcsartifact_fetcher.go

```diff
@@ -97,31 +107,41 @@ func newGCSJobSource(src string) (*gcsJobSource, error) {
 }
 
 // Artifacts lists all artifacts available for the given job source
-func (af *GCSArtifactFetcher) artifacts(src jobSource) []string {
+func (af *GCSArtifactFetcher) artifactNames(src string) ([]string, error) {
+	gcsSrc, err := newGCSJobSource(src)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get GCS job source from %s: %v", src, err)
+	}
 	listStart := time.Now()
-	bucketName, prefix := extractBucketPrefixPair(src.jobPath())
-	artifacts := []string{}
+	bucketName, prefix := extractBucketPrefixPair(gcsSrc.jobPath())
+	artifactNames := []string{}
 	bkt := af.client.Bucket(bucketName)
 	q := storage.Query{
 		Prefix:   prefix,
 		Versions: false,
 	}
 	objIter := bkt.Objects(context.Background(), &q)
-	for {
+	done := false
+	for i := 0; i < maxTries; i++ {
 		oAttrs, err := objIter.Next()
 		if err == iterator.Done {
+			done = true
 			break
 		}
 		if err != nil {
-			logrus.WithFields(fieldsForJob(src)).WithError(err).Error("Error accessing GCS artifact.")
+			logrus.WithFields(fieldsForJob(gcsSrc)).WithError(err).Error("Error accessing GCS artifact.")
 			continue
 		}
-		artifacts = append(artifacts, strings.TrimPrefix(oAttrs.Name, prefix))
+		artifactNames = append(artifactNames, strings.TrimPrefix(oAttrs.Name, prefix))
 
 	}
+	if !done {
+		return artifactNames, fmt.Errorf("Gave up after %d tries", maxTries)
+	}
+
 	listElapsed := time.Since(listStart)
-	logrus.WithField("duration", listElapsed).Infof("Listed %d artifacts.", len(artifacts))
-	return artifacts
```

Comment:
Can you try to split out rote re-names in your commits? It's really hard to review the substance of this PR when so many diff lines are `s/oldName/newName/g`